### PR TITLE
Added a AUTHORS.md file and made performance improvements.

### PR DESCRIPTION
**Pull request by ALEXWARELLC**
# Merge Notes | **LocalAdmin** v 2.5.11

1. Added 'AUTHORS.md'. This file is designed to store all the names of everyone who has contributed to LocalAdmin. When someone creates their first pull request, they should add their name to the table. New contributors should only add their names if the same pull request includes approved code.
2. Reworked the 'OSPlatform' detection system. LocalAdmin now uses a string as storage for the OSPlatform. This allows for switch statements to be used instead of calling IsOSPlatform very time. _This saves on performance as your not calling the function which only really needs to be called once._
3. Updated 'CurrentConfigPath' to use _Path.Combine()_ instead of string interpolation. Path.Combine is designed to create paths that File Readers can support.
4. Added .vscode to .gitignore as it is specific to my PC and contains my specific configuration in VSCode.

ALEXWARELLC

### CODE REVIEW

**Comment by ALEXWARELLC** [comment]
Some of the commits are bull and don't do anything but it was because Git was having a stroke while trying to use VSCode.

**Comment by ALEXWARELLC** [comment]
Also, can i get global tag in game plz.

File: Core/LocalAdmin.cs [hunk 1]
```diff
@@ -27,11 +27,12 @@ namespace LocalAdmin.V2.Core;
     * DarkGreen - success
     * Blue - normal SCPSL log
 */
-
 public sealed class LocalAdmin : IDisposable
 {
     public const string VersionString = "2.5.11";
 
+    public static ushort DefaultPort = 7777;
```
Comment (zabszk) [inline Core/LocalAdmin.cs:34]:
```suggestion
    private const ushort DefaultPort = 7777;
```

File: Core/LocalAdmin.cs [hunk 2]
```diff
@@ -459,15 +465,14 @@ internal async Task Start(string[] args)
             }
             else
             {
-                CurrentConfigPath =
-                    $"{PathManager.GameUserDataRoot}config{Path.DirectorySeparatorChar}{GamePort}{Path.DirectorySeparatorChar}config_localadmin.txt";
-
+                //CurrentConfigPath = $"{}config{Path.DirectorySeparatorChar}{GamePort}{Path.DirectorySeparatorChar}config_localadmin.txt";
```
Comment (zabszk) [inline Core/LocalAdmin.cs:468]:
```suggestion
```

File: Core/LocalAdmin.cs [hunk 3]
```diff
@@ -459,15 +465,14 @@ internal async Task Start(string[] args)
...
+                CurrentConfigPath = Path.Combine(PathManager.GameUserDataRoot, "config", GamePort.ToString(), "config_localadmin.txt");
                 if (File.Exists(CurrentConfigPath))
                     Configuration = Config.DeserializeConfig(await File.ReadAllLinesAsync(CurrentConfigPath, Encoding.UTF8));
                 else
                 {
-                    CurrentConfigPath = $"{PathManager.GameUserDataRoot}config{Path.DirectorySeparatorChar}config_localadmin_global.txt";
-
+                    CurrentConfigPath = Path.Combine(PathManager.GameUserDataRoot, "config", GamePort.ToString(), "config_localadmin_global.txt");
+                    //CurrentConfigPath = $"{PathManager.GameUserDataRoot}config{Path.DirectorySeparatorChar}config_localadmin_global.txt";
```
Comment (zabszk) [inline Core/LocalAdmin.cs:475]:
```suggestion
```

**Review by zabszk — CHANGES_REQUESTED** [review]

**Comment by zabszk** [comment]
Instead of using `RuntimeInformation.IsOSPlatform` you introduced a string variable and switch statements. Take a look at the screenshots below how the `IsOSPlatform` is actually implemented. It's basically comparing a string to a const.

The only performance improvement in your implementation comes from using a switch statement instead of a couple of if statements, but the impact is negligible in this case.

Actually the fastest way to check the OS seems to be using `OperatingSystem.IsLinux()`, `OperatingSystem.IsWindows()` since it's just returning a bool directly. I attached a screenshot how it's implemented as well.








When it comes to the "SCP: SL Developer" global badge - the badge is only for developers who work for us and develop the SCP:SL game itself.

**Comment by ALEXWARELLC** [comment]
Completed changes as requested.

File: Core/LocalAdmin.cs (hunk 1 above)
Comment (zabszk) [reply Core/LocalAdmin.cs:34]: This is back public static after your latest changes.

File: Core/LocalAdmin.cs [hunk 4]
```diff
@@ -123,25 +127,22 @@ internal async Task Start(string[] args)
...
             ConsoleUtil.WriteLine($"Welcome to LocalAdmin version {VersionString}!", ConsoleColor.Red);
             ConsoleUtil.WriteLine("Can't obtain a valid user home directory path!", ConsoleColor.Red);
-
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
-            {
-                ConsoleUtil.WriteLine("Make sure to export a valid path, for example using command: export HOME=/home/username-here", ConsoleColor.Red);
-                ConsoleUtil.WriteLine("You may want to add that command to the top of ~/.bashrc file and restart the terminal session to avoid having to enter that command every time.", ConsoleColor.Red);
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            //If statements will expose future performace issues. switching "CurrentPlatform" (string) will reduce the startup time.
```
Comment (zabszk) [inline Core/LocalAdmin.cs:130]:
```suggestion
```
Comment no longer relevant.

File: Core/LocalAdmin.cs [hunk 5]
```diff
@@ -123,25 +127,22 @@ internal async Task Start(string[] args)
...
+            if (OperatingSystem.IsWindows())
             {
                 ConsoleUtil.WriteLine("Such error should never occur on Windows.", ConsoleColor.Red);
                 ConsoleUtil.WriteLine("Open issue on the LocalAdmin GitHub repository (https://github.com/northwood-studios/LocalAdmin-V2/issues) or contact our technical support!", ConsoleColor.Red);
+            }else if (OperatingSystem.IsLinux())
```
Comment (zabszk) [inline Core/LocalAdmin.cs:135]:
```suggestion
            }
            else if (OperatingSystem.IsLinux())
```

File: Core/LocalAdmin.cs [hunk 6]
```diff
@@ -99,12 +99,16 @@ internal enum HeartbeatStatus : byte
...
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)){
             _scpslExecutable = "SCPSL.exe";
-        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        }
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)){
             _scpslExecutable = "SCPSL.x86_64";
+        }
         else
         {
+            //This occurs if the platform is not a Linux or Windows Machine.
```
Comment (zabszk) [inline Core/LocalAdmin.cs:111]: 1. Comments are no longer relevant.
2. For one line if statement content we don't use braces.
3. Can you use "OperatingSystem.is..." here, in other places in this file (around lines 600 and 870) and in other files (IO/Config.cs and IO/PathManager.cs) as well?

**Review by zabszk — CHANGES_REQUESTED** [review]
I reviewed your last changes, left some comments and after resolving these changes the PR should be ready to be merged.

**Comment by ALEXWARELLC** [comment]
Done. PR can now be merged. :)

File: Core/LocalAdmin.cs [hunk 7]
```diff
@@ -459,15 +458,12 @@ internal async Task Start(string[] args)
...
-                    $"{PathManager.GameUserDataRoot}config{Path.DirectorySeparatorChar}{GamePort}{Path.DirectorySeparatorChar}config_localadmin.txt";
-
+                CurrentConfigPath = Path.Combine(PathManager.GameUserDataRoot, "config", GamePort.ToString(), "config_localadmin.txt");
                 if (File.Exists(CurrentConfigPath))
                     Configuration = Config.DeserializeConfig(await File.ReadAllLinesAsync(CurrentConfigPath, Encoding.UTF8));
                 else
                 {
-                    CurrentConfigPath = $"{PathManager.GameUserDataRoot}config{Path.DirectorySeparatorChar}config_localadmin_global.txt";
-
+                    CurrentConfigPath = Path.Combine(PathManager.GameUserDataRoot, "config", GamePort.ToString(), "config_localadmin_global.txt");
```
Comment (zabszk) [inline Core/LocalAdmin.cs:466]:
```suggestion
                    CurrentConfigPath = Path.Combine(PathManager.GameUserDataRoot, "config", "config_localadmin_global.txt");
```

**Comment by zabszk** [comment]
One issue you marked as resolved wasn't really resolved. I marked it as unresolved.

Right now I'm discussing with other maintainers whether we want to have AUTHORS.md file. List of all contributors is automatically generated by GitHub (Insights tab) and we already credit every contributor in each release of LocalAdmin. I don't really see a point in duplicating the list in another place.

Btw. in the future please make separate Pull Requests for changes that are completely not related to each other. It makes the review process easier and faster.

File: AUTHORS.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# LocalAdmin
```
Comment (MichalPetryka) [inline AUTHORS.md:1]: This file doesn't make sense, source code doesn't need to include the contributors.

File: Core/LocalAdmin.cs [hunk 9]
```diff
@@ -99,10 +99,12 @@ internal enum HeartbeatStatus : byte
 
     internal LocalAdmin()
     {
-        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)){
```
Comment (MichalPetryka) [inline Core/LocalAdmin.cs:102]: We don't do same line `{`.

**Review by MichalPetryka — CHANGES_REQUESTED** [review]

File: Core/LocalAdmin.cs (hunk 9 above)
Comment (ALEXWARELLC) [reply Core/LocalAdmin.cs:102]: This is being removed as it is a 1 liner.

File: AUTHORS.md (hunk 8 above)
Comment (ALEXWARELLC) [reply AUTHORS.md]: Deleted.

File: .gitignore [hunk 10]
```diff
@@ -263,4 +263,6 @@ __pycache__/
 *.pyc
 
 # JSON Schemas
-JSON/Schemas
\ No newline at end of file
+JSON/Schemas
```
[the remark this answers is not in the document]
Comment (ALEXWARELLC) [reply .gitignore:266]: What is this refering to?

**Comment by ALEXWARELLC** [comment]
> One issue you marked as resolved wasn't really resolved. I marked it as unresolved.
> 
> Right now I'm discussing with other maintainers whether we want to have AUTHORS.md file. List of all contributors is automatically generated by GitHub (Insights tab) and we already credit every contributor in each release of LocalAdmin. I don't really see a point in duplicating the list in another place.
> 
> Btw. in the future please make separate Pull Requests for changes that are completely not related to each other. It makes the review process easier and faster.

Ok. Imma request another review.

**Review by zabszk — APPROVED** [review]

**Review by MichalPetryka — APPROVED** [review]

**Comment by ALEXWARELLC** [comment]
Quick Question. Why are the Linux Builds failing?

**Comment by ALEXWARELLC** [comment]
The windows builds work just fine but the linux build in the CI pipeline fail without even starting.

**Comment by zabszk** [comment]
> Quick Question. Why are the Linux Builds failing?

GitHub dropped support for Ubuntu 18.04 builds. I just merged a PR changing it to 20.04 LTS. We are using older versions for compatibility reasons (some people host servers on older linux versions).

**Pull request merged**